### PR TITLE
planner: unsigned pk cannot be pushed as index column

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

It's a hard work to keep data in order if the index range contains unsigned pk since the pk is the last column not the first column.

### What is changed and how it works?

Disabled it when the pk is unsigned.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/mock.go [hunk 1]
```diff
@@ -263,6 +263,51 @@ func MockTable() *model.TableInfo {
 	return table
 }
 
+// MockTableT2 is only used for plan related tests.
+func MockTableT2() *model.TableInfo {
```
Comment (zz-jason) [inline planner/core/mock.go]: when should we use `MockTableT` and `MockTableT2`?

Comment (winoros) [reply planner/core/mock.go]: One holds a signed pk and one holds a unsigned pk. Maybe there'll be a table hold not integer pk in the future.

Comment (lamxTyler) [reply planner/core/mock.go]: Can we put this information in the name? Like `MockSignedPKTable`, we could tell the difference without looking into the code.

Comment (winoros) [reply planner/core/mock.go]: Addressed.

**Pull request merged**